test/integration/instance_config_test.go

```diff
@@ -0,0 +1,431 @@
+package integration
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	. "github.com/linode/linodego"
+)
+
+func setupVPCWithSubnetWithInstance(
+	t *testing.T,
+	fixturesYaml string,
+	modifiers ...instanceModifier,
+) (
+	*Client,
+	*VPC,
+	*VPCSubnet,
+	*Instance,
+	*InstanceConfig,
+	func(),
+	error,
+) {
+	t.Helper()
+	client, instance, instanceConfig, instanceTeardown, err := setupInstanceWithoutDisks(
+		t,
+		fixturesYaml,
+		modifiers...,
+	)
+	if err != nil {
+		if instanceTeardown != nil {
+			instanceTeardown()
+		}
+		t.Fatal(err)
+	}
+
+	vpc, vpcSubnet, vpcWithSubnetTeardown, err := createVPCWithSubnet(
+		t,
+		client,
+		func(client *Client, options *VPCCreateOptions) {
+			options.Region = instance.Region
+		},
+	)
+	if err != nil {
+		t.Error(err)
+	}
+
+	teardownAll := func() {
+		instanceTeardown()
+		vpcWithSubnetTeardown()
+	}
+	return client, vpc, vpcSubnet, instance, instanceConfig, teardownAll, err
+}
+
+func setupInstanceWith3Interfaces(t *testing.T, fixturesYaml string) (
+	*Client,
+	*VPC,
+	*VPCSubnet,
+	*Instance,
+	*InstanceConfig,
+	func(),
+) {
+	t.Helper()
+	client, vpc, vpcSubnet, instance, config, teardown, err := setupVPCWithSubnetWithInstance(
+		t,
+		fixturesYaml,
+		func(client *Client, opts *InstanceCreateOptions) {
+			opts.Region = getRegionsWithCaps(t, client, []string{"VPCs"})[0]
+		},
+	)
+	if err != nil {
+		if teardown != nil {
+			teardown()
+		}
+		t.Fatal(err)
+	}
+
+	updateConfigOpts := config.GetUpdateOptions()
+	updateConfigOpts.Interfaces = []InstanceConfigInterfaceCreateOptions{
+		{
+			Purpose: InterfacePurposePublic,
+		},
+		{
+			Purpose: InterfacePurposeVLAN,
+			Label:   "testvlan",
+		},
+		{
+			Purpose:  InterfacePurposeVPC,
+			SubnetID: &vpcSubnet.ID,
+			IPv4: &VPCIPv4{
+				NAT1To1: "any",
+			},
+		},
+	}
+	config, err = client.UpdateInstanceConfig(context.Background(), instance.ID, config.ID, updateConfigOpts)
+	if err != nil {
+		teardown()
+		t.Fatal(err)
+	}
+
+	return client, vpc, vpcSubnet, instance, config, teardown
+}
+
+func TestInstance_ConfigInterfaces_AppendDelete(t *testing.T) {
```

Comment:
After re-running the fixtures, I noticed this test was failing when running `make SKIP_LINT=1 test` with a missing interaction error. As far as I can tell, this is because [fixturesTeardown](https://github.com/linode/linodego/blob/77af41ef44271e8302df4829764c9eb26d9450cf/test/integration/instances_test.go#L537) is called before the [VPC teardown] is called, which is causing the VPC deletion to not be saved in the fixtures file.

Reply:
As a not-so-pretty workaround, I was able to fix this issue by swapping the order of these two lines: https://github.com/linode/linodego/blob/b38fbd67135b16e6cd75ac9e59c3159d5daee5cc/test/integration/instance_config_test.go#L49-L50

Reply:
Thank you! I think we have resolved this.